cli/tests/pcluster/config/test_section_cluster.py

```diff
@@ -109,13 +112,87 @@ def test_cluster_section_from_cfn(mocker, cfn_params_dict, expected_section_dict
             None,
             "'invalid_key.*,invalid_key.*' are not allowed in the .* section",
         ),
+        # CLUSTER_HIT parameters must not be allowed in CLUSTER_SIT
+        (
+            {"cluster default": {"queue_settings": "fake_value"}},
+            None,
+            "'queue_settings' is not allowed in the .* section",
+        ),
+        (
+            {"cluster default": {"default_queue": "fake_value"}},
+            None,
+            "'default_queue' is not allowed in the .* section",
+        ),
     ],
 )
-def test_cluster_section_from_file(mocker, config_parser_dict, expected_dict_params, expected_message):
+def test_sit_cluster_section_from_file(mocker, config_parser_dict, expected_dict_params, expected_message):
     utils.set_default_values_for_required_cluster_section_params(
         config_parser_dict.get("cluster default"), only_if_not_present=True
     )
-    utils.assert_section_from_file(mocker, CLUSTER, config_parser_dict, expected_dict_params, expected_message)
+    utils.assert_section_from_file(mocker, CLUSTER_SIT, config_parser_dict, expected_dict_params, expected_message)
+
+
+@pytest.mark.parametrize(
+    "config_parser_dict, expected_dict_params, expected_message",
+    [
+        # default
+        (
+            {"cluster default": {}},
+            {"additional_iam_policies": [], "architecture": None, "scheduler": "slurm", "base_os": "alinux2"},
+            None,
+        ),
+        # right value
+        (
+            {"cluster default": {"key_name": "test"}},
+            {
+                "key_name": "test",
+                "additional_iam_policies": [],
+                "architecture": None,
+                "scheduler": "slurm",
+                "base_os": "alinux2",
+            },
+            None,
+        ),
+        (
+            {"cluster default": {"base_os": "alinux"}},
+            {"base_os": "alinux", "additional_iam_policies": [], "architecture": None, "scheduler": "slurm"},
+            None,
+        ),
+        # invalid value
+        ({"cluster default": {"base_os": "wrong_value"}}, {}, "has an invalid value"),
+        # invalid key
+        ({"cluster default": {"invalid_key": "fake_value"}}, {}, "'invalid_key' is not allowed in the .* section"),
+        # CLUSTER_SIT parameters must not be allowed in CLUSTER_HIT
+        (
+            {"cluster default": {"placement_group": "fake_value"}},
+            {},
+            "'placement_group' is not allowed in the .* section",
+        ),
+        ({"cluster default": {"placement": "ondemand"}}, {}, "'placement' is not allowed in the .* section"),
+        (
+            {"cluster default": {"compute_instance_type": "t2.micro"}},
+            {},
+            "'compute_instance_type' is not allowed in the .* section",
+        ),
+        ({"cluster default": {"initial_queue_size": 0}}, {}, "'initial_queue_size' is not allowed in the .* section"),
```

Comment:
Let's remember to adapt the documentation for each parameters, specifyin when it can be used or not.

Reply:
+ @HLeeHart